### PR TITLE
Added Pure Python mode to Calling C functions tutorial

### PR DESCRIPTION
**Pull request by matusvalo**
Partial fix of #4187.

### CODE REVIEW

File: docs/src/tutorial/external.rst [hunk 1]
```diff
@@ -15,7 +18,17 @@ For example, let's say you need a low-level way to parse a number from
 a ``char*`` value.  You could use the ``atoi()`` function, as defined
```
Comment (0dminnimda) [inline docs/src/tutorial/external.rst:18]: In pure python mode, you need to use `cython.p_char`

Comment (matusvalo) [reply docs/src/tutorial/external.rst:18]: I understand the text in a way that it discuss in context of C language (e.g. it mentions also `stdlib.h` header file.

File: docs/src/tutorial/external.rst [hunk 2]
```diff
@@ -103,7 +138,19 @@ You can now make it clear which of the two arguments does what in
...
 
-.. literalinclude:: ../../examples/tutorial/external/keyword_args_call.pyx
+.. tabs::
+
+    .. group-tab:: Pure Python
+
+        .. literalinclude:: ../../examples/tutorial/external/keyword_args_call.pxd
+            :caption: keyword_args_call.pxd
+        .. literalinclude:: ../../examples/tutorial/external/keyword_args_call.py
+            :caption: keyword_args_call.py
```
Comment (scoder) [inline docs/src/tutorial/external.rst:148]: I don't think it's didactically helpful to use the module's own .pxd file for the declaration. At least not without an elaborate explanation of how it's different from other .pxd files and what it does to your module and its namespace.

You can just name it `strstr.pxd`. That's all it provides, and it (hopefully) suggests naming .pxd files after what they provide.

Comment (matusvalo) [reply docs/src/tutorial/external.rst]: done.

**Comment by matusvalo** [comment]
I have implemented all comments. Could you please do a final review?

File: docs/src/tutorial/external.rst [hunk 3]
```diff
@@ -83,6 +115,9 @@ This allows the C declaration to be reused in other Cython modules,
 while still providing an automatically generated Python wrapper in
 this specific module.
 
+.. note:: External declarations must be placed in .pxd file in Pure
```
Comment (0dminnimda) [inline docs/src/tutorial/external.rst:118]: ``.pxd``

Comment (0dminnimda) [inline docs/src/tutorial/external.rst:118]:
```suggestion
.. note:: External declarations must be placed in ``.pxd`` file in Pure
```

**Comment by 0dminnimda** [comment]
Everything seems good, code and text, except you may want to somehow align pyx with pxd and py. Like add some spaces in needed places so with change lines from two files will align with one. Idk if it's possible but you can try to make them as aligned

**Comment by matusvalo** [comment]
> except you may want to somehow align pyx with pxd and py (keyword_args_call). 

done. Still not best but now the lines should match.

**Comment by 0dminnimda** [comment]
> > except you may want to somehow align pyx with pxd and py (keyword_args_call).
> 
> done. Still not best but now the lines should match.

If lines are matched already good.
Hmm I thought about adding pxd first and then aligning py lines with rest of pyx. But current alignment is good too.

**Comment by scoder** [comment]
Thanks!

**Comment by mattip** [comment]
This work is very much appreciated. It makes the pure-python mode much clearer and will aid its adoption, and cleans up some corners in the documentation. Thanks to all the reviewers and the contributors (from a bystander) for your efforts.

**Comment by scoder** [comment]
> This work is very much appreciated. It makes the pure-python mode much clearer and will aid its adoption, and cleans up some corners in the documentation. Thanks to all the reviewers and the contributors (from a bystander) for your efforts.

+100. It even helps us find issues with the current implementation and thus improves the overall user experience with it, not just the documentation.

**Comment by 0dminnimda** [comment]
> +100. It even helps us find issues with the current implementation and thus improves the overall user experience with it, not just the documentation.

I think pure python mode benefits alot from new issues with requests to equate both versions. In the end it will allow to fully develop under pure version.

**Pull request merged**